src/components/ColorPicker.tsx

```diff
@@ -42,14 +49,90 @@ const ColorSwatch: FC<{
   const dispatch = useDispatch()
   const colors = useSelector(themeColors)
   const fontSize = useSelector(state => state.fontSize)
-  const selected =
-    (color && cursorStyle?.color === color) || (backgroundColor && cursorStyle?.backgroundColor === backgroundColor)
+  const currentThoughtId = useSelector(state => !!state.cursor && head(state.cursor))
+  const currentThoughtValue = useSelector(
+    state => (!!state.cursor && getThoughtById(state, head(state.cursor))?.value) || '',
+  )
+  const simplePath = useSelector(state => simplifyPath(state, state.cursor!))
   size = size || fontSize * 1.2
 
+  /** A function that adds an alpha channel to a hex color. */
+  const addAlphaToHex = (hex: string) => {
+    if (hex.length === 7) return hex + 'ff'
+    return hex
+  }
+
+  const selected = useMemo(() => {
```

Comment:
Rather than use `useMemo` with several `useSelector` values as dependencies, wrap all this logic in a single `useSelector`. Then you have access to a fresh State and the component is only re-rendered when the `selected` boolean value changes.